crates/bevy_ecs/src/system/combinator.rs

```diff
@@ -300,3 +300,25 @@ where
         b(value)
     }
 }
+
+pub type ProductSystem<SystemA, SystemB> = CombinatorSystem<CartesianProduct, SystemA, SystemB>;
+
+pub struct CartesianProduct;
+
+impl<In, A, B> Combine<A, B> for CartesianProduct
+where
+    In: Copy,
```

Comment:
This bound feels incorrect: if we're joining the input we should just be able to pass it by value all the way through.

Reply:
Yes, I've tried to remove the Copy, but wasn't able to